src/main/java/seedu/exchangecoursemapper/parser/CourseValidator.java

```diff
@@ -0,0 +1,111 @@
+package seedu.exchangecoursemapper.parser;
+
+import seedu.exchangecoursemapper.constants.Logs;
+import seedu.exchangecoursemapper.exception.Exception;
+
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static seedu.exchangecoursemapper.constants.JsonKey.PU_COURSE_NAME_KEY;
+import static seedu.exchangecoursemapper.constants.JsonKey.PU_COURSE_CODE_KEY;
+import static seedu.exchangecoursemapper.constants.JsonKey.NUS_COURSE_NAME_KEY;
+import static seedu.exchangecoursemapper.constants.JsonKey.NUS_COURSE_CODE_KEY;
+import static seedu.exchangecoursemapper.constants.JsonKey.COURSES_ARRAY_LABEL;
+import static seedu.exchangecoursemapper.constants.Messages.LINE_SEPARATOR;
+import static seedu.exchangecoursemapper.constants.Messages.LIST_RELEVANT_PU;
+
+public class CourseValidator {
```

Comment:
I like that you extracted out the functions of validating the user's input for adding courses! It ensures that the code upholds the Single Responsibility Principle 👍